File: 2610_matrix/Matrix.cpp
class Solution {
public:
    vector<vector<int>> findMatrix(vector<int>& nums) {
        vector<int> frequency(nums.size() + 1);
        vector<vector<int>> result;

        for (int i : nums) {
            if (frequency[i] >= result.size()) {
                result.push_back({});
            }

            result[frequency[i]].push_back(i);
            frequency[i]++;
        }

        return result;
    }
};

File: 2610_matrix/Matrix.java
class Solution {
    public List<List<Integer>> findMatrix(int[] nums) {
        List<List<Integer>> res = new ArrayList<>();
        int[] frequency = new int[nums.length + 1];

        for (int i : nums) {
            if (frequency[i] >= res.size()) {
                res.add(new ArrayList<>());
            }

            res.get(frequency[i]).add(i);
            frequency[i]++;
        }

        return res;
    }
}